src/loaders/mod_load.c

```diff
@@ -756,6 +790,12 @@ static int mod_load(struct module_data *m, HIO_HANDLE *f, const int start)
 	}
 
 #ifndef LIBXMP_CORE_PLAYER
+	/* First pattern of Software Visions DMF is word flipped (see above). */
+	if (tracker_id == TRACKER_SOFTWAREVISIONS && i == 0) {
+		assert(mod->chn == 4);
```

Comment:
If we ever hit this, we'll face unexpected aborts from the library. The cmake build system defines NDEBUG in release builds, the MSVC project files always seem to define it, but autotools and other ones don't. We should either (i) add an `--enable-debug` switch to autotools configury (too much work not related to this patch), or (ii) adopt what we do on top of vorbis.h and miniz.h more globally - e.g. in common.h. Or, (iii) you can conditionalize the assert here to DEBUG builds, do an error-return with a `D_()` message or something.

Reply:
(if it were me, I'd lean to the 3rd option..)

Reply:
I forgot we don't really support this right now; I replaced it with reusing the pattern length and a comment instead.

Reply:
P.S.: Not worth bailing out `if (mod->chn != 4)`?

Reply:
> P.S.: Not worth bailing out `if (mod->chn != 4)`?

Nope, that was just to guard the 1024. It uses the calculated pattern length now so it's safe.